code/game/machinery/computer/camera.dm

```diff
@@ -3,201 +3,47 @@
-			return 0
-
-		A.eyeobj.setLoc(get_turf(C))
-		A.client.eye = A.eyeobj
-		return 1
+	var/mapping = 0//For the overview file, interesting bit of code.
+	var/list/network = list()
 
-	if (!C.can_use() || user.stat || (get_dist(user, src) > 1 || user.machine != src || user.blinded || !( user.canmove ) && !istype(user, /mob/living/silicon)))
-		return 0
-	set_current(C)
-	user.reset_view(current_camera)
-	check_eye(user)
-	return 1
+	var/datum/tgui_module/camera/camera
 
-/obj/machinery/computer/security/relaymove(mob/user,direct)
-	var/turf/T = get_turf(current_camera)
-	for(var/i; i < 10; i++)
-		T = get_step(T, direct)
-	jump_on_click(user, T)
+/obj/machinery/computer/security/Initialize()
+	. = ..()
+	if(!LAZYLEN(network))
+		network = get_default_networks()
+	camera = new(src, network)
 
-//Camera control: moving.
-/obj/machinery/computer/security/proc/jump_on_click(var/mob/user,var/A)
-	if(user.machine != src)
-		return
-	var/obj/machinery/camera/jump_to
-	if(istype(A,/obj/machinery/camera))
-		jump_to = A
-	else if(ismob(A))
-		if(ishuman(A))
-			jump_to = locate() in A:head
-		else if(isrobot(A))
-			jump_to = A:camera
-	else if(isobj(A))
-		jump_to = locate() in A
-	else if(isturf(A))
-		var/best_dist = INFINITY
-		for(var/obj/machinery/camera/camera in get_area(A))
-			if(!camera.can_use())
-				continue
-			if(!can_access_camera(camera))
-				continue
-			var/dist = get_dist(camera,A)
-			if(dist < best_dist)
-				best_dist = dist
-				jump_to = camera
-	if(isnull(jump_to))
-		return
-	if(can_access_camera(jump_to))
-		switch_to_camera(user,jump_to)
+/obj/machinery/computer/security/proc/get_default_networks()
+	. = using_map.station_networks.Copy()
 
-/obj/machinery/computer/security/process()
-	if(cache_id != camera_repository.camera_cache_id)
-		cache_id = camera_repository.camera_cache_id
-		SSnanoui.update_uis(src)
+/obj/machinery/computer/security/Destroy()
+	QDEL_NULL(camera)
+	return ..()
 
-/obj/machinery/computer/security/proc/can_access_camera(var/obj/machinery/camera/C)
-	var/list/shared_networks = src.network & C.network
-	if(shared_networks.len)
-		return 1
-	return 0
+/obj/machinery/computer/security/tgui_interact(mob/user, datum/tgui/ui = null)
+	camera.tgui_interact(user, ui)
 
-/obj/machinery/computer/security/proc/set_current(var/obj/machinery/camera/C)
-	if(current_camera == C)
+/obj/machinery/computer/security/attack_hand(mob/user)
+	add_fingerprint(user)
+	if(stat & (BROKEN|NOPOWER))
 		return
+	tgui_interact(user)
 
-	if(current_camera)
-		reset_current()
-
-	src.current_camera = C
-	if(current_camera)
-		current_camera.camera_computers_using_this.Add(src)
-		update_use_power(USE_POWER_ACTIVE)
-		var/mob/living/L = current_camera.loc
-		if(istype(L))
-			L.tracking_initiated()
+/obj/machinery/computer/security/attack_ai(mob/user)
+	to_chat(user, "<span class='notice'>You realise its kind of stupid to access a camera console when you have the entire camera network at your metaphorical fingertips.</span>")

```

Comment:
This whole proc was created on Paradise out of a problem they immediately had where the AI would just have one of it's robots build like 20 camera consoles so that it could watch a ton of cameras at once.